File: code/CPP_vs_Python_run_time.cpp
#include <iostream>
#include <chrono> // Include the chrono library for timing

using namespace std;
using namespace std::chrono; // Use the chrono namespace for convenience

int main()
{
    // Start the timer
    auto start = high_resolution_clock::now();

    int n = 0;
    while (n < 1'000'000'000)
        n++;

    // Stop the timer
    auto stop = high_resolution_clock::now();

    // Calculate the duration
    auto duration = duration_cast<milliseconds>(stop - start);

    // Output the result and the time taken
    cout << "Final value of n: " << n << endl;
    cout << "Time taken: " << duration.count() << " milliseconds" << endl;

    return 0;
}

File: code/CPP_vs_Python_run_time.py
import time

# Start the timer
start_time = time.time()

n = 0
while n < 1_000_000_000:
    n += 1

# Stop the timer
end_time = time.time()

# Calculate the duration
duration = end_time - start_time

# Output the result and the time taken
print(f"Final value of n: {n}")
print(f"Time taken: {duration:.4f} seconds")